pkg/inputs/snmp/disco.go

```diff
@@ -360,6 +361,27 @@ func addDevices(ctx context.Context, foundDevices map[string]*kt.SnmpDeviceConfi
 		}
 	}
 
+	// Remove any devices which havn't been up long enough if this is set.
+	if !isTest {
+		removeNum := conf.Global.PurgeDevices
+		for dip, d := range conf.Devices {
+			if d.PurgeDevice != 0 { // Let this override the global settings.
+				removeNum = d.PurgeDevice
+			}
+
+			if removeNum > 0 && pollDuration > 0 {
```

Comment:
> You can also set this to purge_after_num: -1 to pin a device and never delete it.

I wonder if we specifically code that behavior here. The way this is now, any non-positive value would technically work to keep a device forever. Just thinking that knowledge will get lost over time unless it's actually in code

Reply:
Yeah, good idea to make this this explicit. Now will throw an error if value isn't -1,0,1 to MAX_INT